src/main/java/com/myroslav/cosmickitties/service/implementation/CustomerService.java

```diff
@@ -0,0 +1,78 @@
+package com.myroslav.cosmickitties.service.implementation;
+
+import com.myroslav.cosmickitties.domain.Customer;
+import com.myroslav.cosmickitties.dto.CustomerDTO;
+import com.myroslav.cosmickitties.exception.BadRequestException;
+import com.myroslav.cosmickitties.exception.ResourceNotFoundException;
+import com.myroslav.cosmickitties.mapper.CustomerMapper;
+import com.myroslav.cosmickitties.repository.CustomerRepository;
+import com.myroslav.cosmickitties.service.abstraction.ICustomerService;
+import jakarta.transaction.Transactional;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional
+public class CustomerService implements ICustomerService {
+
+    private final CustomerRepository repo;
+    private final CustomerMapper mapper;
+
+    public CustomerService(CustomerRepository repo, CustomerMapper mapper) {
+        this.repo = repo;
+        this.mapper = mapper;
+    }
+
+    @Override
+    public CustomerDTO create(CustomerDTO dto) {
+        if (dto.getEmail() != null && repo.existsByEmail(dto.getEmail())) {
+            throw new IllegalArgumentException("Email already exists");
```

Comment:
exceptions should be granular. Tha'ts sth that you are also defining on db level, so you should try to save to db, and than falback to the data integrity exception